common/rootfs/docker-entrypoint.sh

```diff
@@ -8,6 +8,20 @@ bashio::log.info "Preparing to start..."
 # - https://github.com/zigbee2mqtt/hassio-zigbee2mqtt/issues/387
 bashio::config.require 'data_path'
 
+export ZIGBEE2MQTT_DATA="$(bashio::config 'data_path')"
+
+# Migrate configuration to addon specific data path for HA backups, see https://github.com/zigbee2mqtt/hassio-zigbee2mqtt/issues/627
+if [[ "$ZIGBEE2MQTT_DATA" == "/addon_config/zigbee2mqtt" ]] && ! bashio::fs.file_exists "$ZIGBEE2MQTT_DATA/configuration.yaml" && bashio::fs.file_exists "/config/zigbee2mqtt/configuration.yaml"; then
+    bashio::log.info "Migrating configuration from /config/zigbee2mqtt to $ZIGBEE2MQTT_DATA/zigbee2mqtt"
+
+    mkdir -p "$ZIGBEE2MQTT_DATA" || bashio::exit.nok "Could not create $ZIGBEE2MQTT_DATA"
+    cp -r /config/zigbee2mqtt/* $ZIGBEE2MQTT_DATA/ || bashio::exit.nok "Error copying configuration files from /config to /addon_config."
+
+    bashio::log.info "Configuration migrated successfully."
+    bashio::log.info "Deleting old configuration from /config/zigbee2mqtt"
+    rm -rf /config/zigbee2mqtt
```

Comment:
Should we keep the directory for some version to allow downgrades? Or does this directory get restored when you restore a backup?

Reply:
A full Home Assistant backup would restore the folder, just restoring the addon would not. I wasn't sure whether to keep it or delete the folder, but I fear keeping it will result in users just upgrading without reading the release notes and then wondering why their settings don't get applied/the folder doesn't reflect changes/logs not getting stored etc...

Reply:
I see, so we should be a bit cautious with this since you cannot downgrade anymore. Maybe release this together with z2m 2.0.0? (https://github.com/Koenkk/zigbee2mqtt/discussions/24198)

Reply:
Is there an estimated timeframe for 2.0? 

Maybe it would make sense to get just the additional mount point merged now so it's possible to manually configure the addon to use the addon_config dir and then update the default/add the automatic migration for 2.0?

Reply:
> Is there an estimated timeframe for 2.0?

In a few months (you can track progress here: https://github.com/Koenkk/zigbee2mqtt/discussions/24198)


> Maybe it would make sense to get just the additional mount point merged now so it's possible to manually configure the addon to use the addon_config dir and then update the default/add the automatic migration for 2.0?

Sounds good!